File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max_profit = 0, price = prices[0];
        for(int i=1; i<prices.size(); i++) {
            max_profit = max(max_profit,prices[i] - price);
            price = min(price,prices[i]);
        }
        return max_profit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit = 0;
        int buy = prices[0];
        for(int i=1; i<prices.length; i++) {
            int sell = prices[i] - buy;
            profit = Math.max(profit,sell);
            buy = Math.min(buy,prices[i]);
        }
        return profit;
    }
}